scripts/cmake-configurator/parser/test_cmakevariable.py

```diff
@@ -0,0 +1,28 @@
+import unittest
+from cmake_row import cmake_row
+from cmake_variable import cmake_variable
+
+comment: str = "Build unit tests for the target arch"
+bool_line: str = "ARCH_GTESTS:BOOL=OFF"
+internal_line: str = "CMAKE_AR:INTERNAL=/usr/bin/ar"
+arch_row: cmake_row = cmake_row(bool_line, 0)
+
+class test_cmake_cache_variable(unittest.TestCase):
+
+   def test_var_creation(self):
+      var = cmake_variable(arch_row, comment)
+      self.assertEqual(var.row.name, "ARCH_GTESTS") \
+         and self.assertEqual(var.val == "OFF") \
+         and self.assertEqual(var.type == "BOOL")
+
+   def test_input_valid_bool(self):
+      var = cmake_variable(arch_row, comment)
+      with self.subTest():
```

Comment:
Cool, but it would be even cooler to check for all 6 cases. See the comment above.